modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/export/stats/JsonContributionStatsExporter.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * @since 22.0.0
+ */
+public class JsonContributionStatsExporter extends AbstractExporter {
+
+    protected static final String STUDIO_COMPONENT_PREFIX = "studio.extensions";
+
+    public JsonContributionStatsExporter(ExporterDescriptor descriptor) {
+        super(descriptor);
+    }
+
+    protected CodeType computeCodeType(ExtensionInfo contrib, List<String> javaTypes, List<String> scriptingTypes) {
+        String xp = contrib.getExtensionPoint();
+        if (javaTypes.contains(xp)) {
+            return CodeType.JAVA;
+        }
+        if (scriptingTypes.contains(xp)) {
+            return CodeType.SCRIPTING;
+        }
+        return CodeType.XML;
+    }
+
+    protected List<String> addAliases(DistributionSnapshot distribution, List<String> xps) {
+        List<String> res = new ArrayList<>();
+        xps.forEach(xp -> {
+            res.add(xp);
+            ExtensionPointInfo xpInfo = distribution.getExtensionPoint(xp);
+            if (xpInfo != null) {
+                res.addAll(xpInfo.getAliases());
+            }
+        });
+        return res;
+    }
+
+    protected List<ContributionStat> computeStats(DistributionSnapshot distribution, SnapshotFilter filter,
+            Map<String, String> properties) {
+        List<String> javaTypes = addAliases(distribution, CodeType.JAVA.getValues(properties, getProperties()));
+        List<String> scriptingTypes = addAliases(distribution,
+                CodeType.SCRIPTING.getValues(properties, getProperties()));
+        return distribution.getContributions()
+                           .stream()
+                           .sorted(Comparator.comparing(ExtensionInfo::getId))
+                           .filter(contrib -> filter == null || filter.accept(contrib))
+                           .map(contrib -> {
+                               ContributionStat stat = new ContributionStat();
+                               stat.setExtensionId(contrib.getId());
+                               String xp = contrib.getExtensionPoint();
+                               stat.setTargetExtensionPointId(xp);
+                               stat.setTargetExtensionPointPresent(distribution.getExtensionPoint(xp) != null);
+                               stat.setNumberOfContributions(Long.valueOf(contrib.getContributionItems().size()));
+                               stat.setCodeType(computeCodeType(contrib, javaTypes, scriptingTypes));
+                               stat.setFromStudio(contrib.getId().startsWith(STUDIO_COMPONENT_PREFIX));
+                               return stat;
+                           })
+                           .collect(Collectors.toList());
+    }
+
+    @Override
+    public void export(OutputStream out, DistributionSnapshot distribution, SnapshotFilter filter,
+            Map<String, String> properties) {
+        List<ContributionStat> stats = computeStats(distribution, filter, properties);
+
+        final ObjectMapper mapper = new ObjectMapper();
```

Comment:
The mapper should be a class constant, as it's expensive to create.

Reply:
Moved the whole writer as a class constant